p2p/switch_test.go

```diff
@@ -443,23 +431,68 @@ func TestSwitchReconnectsToPersistentPeer(t *testing.T) {
 	rp.Start()
 	defer rp.Stop()
 
-	// simulate first time dial failure
 	conf := config.DefaultP2PConfig()
 	conf.TestDialFail = true
-	err = sw.addOutboundPeerWithConfig(rp.Addr(), conf, true)
-	require.NotNil(err)
-
+	err = sw.addOutboundPeerWithConfig(rp.Addr(), conf)
```

Comment:
don't we need to make this peer persistent to preserve the test case?

Reply:
No. Because we'll reconnect anyway if `TestDialFail` is set
https://github.com/tendermint/tendermint/blob/5051a1f7bcda164835592baccee3caf0d02984ec/p2p/switch.go#L653